<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>biocGO: describe relationship between Gene Ontology and Bioconductor • biocGO</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="biocGO: describe relationship between Gene Ontology and Bioconductor">
<meta property="og:description" content="biocGO">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-93043521-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-93043521-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">biocGO</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/biocGO.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/vjcitn/biocGO">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="biocGO_files/header-attrs-2.11/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>biocGO: describe relationship between Gene Ontology and Bioconductor</h1>
                        <h4 class="author">Vincent J. Carey</h4>
                  <a class="author_email" href="mailto:#">stvjc at channing.harvard.edu</a>
      
                              <h4 class="author">Ludwig Geistlinger</h4>
                  <a class="author_email" href="mailto:#">ludwig_geistlinger at hms.harvard.edu</a>
      
                  
            <h4 class="date">September 29, 2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/vjcitn/biocGO/blob/master/vignettes/biocGO.Rmd"><code>vignettes/biocGO.Rmd</code></a></small>
      <div class="hidden name"><code>biocGO.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>Gene Ontology reaches into Bioconductor in many ways. This vignette sketches the key resources and connections.</p>
</div>
<div id="go-db" class="section level1">
<h1 class="hasAnchor">
<a href="#go-db" class="anchor"></a>GO.db</h1>
<p><a href="https://bioconductor.org/packages/GO.db/"><code>GO.db</code></a> is an <em>annotation package</em> used through an R API defined in <code>AnnotationDbi</code>. The package is installable using <code><a href="https://rdrr.io/pkg/BiocManager/man/install.html">BiocManager::install</a></code>.</p>
<div id="the-basic-r-api" class="section level2">
<h2 class="hasAnchor">
<a href="#the-basic-r-api" class="anchor"></a>The basic R API</h2>
<p>Here is how we can acquire some tags and the associated terms.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">GO.db</span><span class="op">)</span>
<span class="va">GO.db</span></code></pre></div>
<pre><code>## GODb object:
## | GOSOURCENAME: Gene Ontology
## | GOSOURCEURL: http://current.geneontology.org/ontology/go-basic.obo
## | GOSOURCEDATE: 2021-02-01
## | Db type: GODb
## | package: AnnotationDbi
## | DBSCHEMA: GO_DB
## | GOEGSOURCEDATE: 2021-Apr14
## | GOEGSOURCENAME: Entrez Gene
## | GOEGSOURCEURL: ftp://ftp.ncbi.nlm.nih.gov/gene/DATA
## | DBSCHEMAVERSION: 2.1</code></pre>
<pre><code>## 
## Please see: help('select') for usage information</code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">lk</span> <span class="op">=</span> <span class="fu">keys</span><span class="op">(</span><span class="va">GO.db</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span>
<span class="va">lk</span></code></pre></div>
<pre><code>##  [1] "GO:0000001" "GO:0000002" "GO:0000003" "GO:0000006" "GO:0000007" "GO:0000009" "GO:0000010"
##  [8] "GO:0000011" "GO:0000012" "GO:0000014"</code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">tt</span> <span class="op">=</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html">select</a></span><span class="op">(</span><span class="va">GO.db</span>, keys<span class="op">=</span><span class="va">lk</span>, columns<span class="op">=</span><span class="st">"TERM"</span><span class="op">)</span></code></pre></div>
<pre><code>## 'select()' returned 1:1 mapping between keys and columns</code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">tt</span></code></pre></div>
<pre><code>##          GOID                                                     TERM
## 1  GO:0000001                                mitochondrion inheritance
## 2  GO:0000002                         mitochondrial genome maintenance
## 3  GO:0000003                                             reproduction
## 4  GO:0000006    high-affinity zinc transmembrane transporter activity
## 5  GO:0000007 low-affinity zinc ion transmembrane transporter activity
## 6  GO:0000009                   alpha-1,6-mannosyltransferase activity
## 7  GO:0000010                trans-hexaprenyltranstransferase activity
## 8  GO:0000011                                      vacuole inheritance
## 9  GO:0000012                               single strand break repair
## 10 GO:0000014       single-stranded DNA endodeoxyribonuclease activity</code></pre>
</div>
<div id="relationships" class="section level2">
<h2 class="hasAnchor">
<a href="#relationships" class="anchor"></a>Relationships</h2>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span><span class="st">"package:GO.db"</span><span class="op">)</span></code></pre></div>
<pre><code>##  [1] "GO"            "GO_dbconn"     "GO_dbfile"     "GO_dbInfo"     "GO_dbschema"   "GO.db"        
##  [7] "GOBPANCESTOR"  "GOBPCHILDREN"  "GOBPOFFSPRING" "GOBPPARENTS"   "GOCCANCESTOR"  "GOCCCHILDREN" 
## [13] "GOCCOFFSPRING" "GOCCPARENTS"   "GOMAPCOUNTS"   "GOMFANCESTOR"  "GOMFCHILDREN"  "GOMFOFFSPRING"
## [19] "GOMFPARENTS"   "GOOBSOLETE"    "GOSYNONYM"     "GOTERM"</code></pre>
<p>The GO{BP,MF,CC}* entities listed here are special R environments that can be queried.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/get.html">get</a></span><span class="op">(</span><span class="st">"GO:0000012"</span>, <span class="va">GOBPCHILDREN</span><span class="op">)</span></code></pre></div>
<pre><code>##            regulates negatively regulates positively regulates                  isa 
##         "GO:1903516"         "GO:1903517"         "GO:1903518"         "GO:1903823" 
##                  isa 
##         "GO:1990396"</code></pre>
<p>The recorded relationships for BP terms are:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">kk</span> <span class="op">=</span> <span class="fu">keys</span><span class="op">(</span><span class="va">GO.db</span><span class="op">)</span>
<span class="va">allc</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/get.html">mget</a></span><span class="op">(</span><span class="va">kk</span>, <span class="va">GOBPCHILDREN</span>, ifnotfound<span class="op">=</span><span class="cn">NA</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">allc</span>, <span class="va">names</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## 
##                  isa negatively regulates              part of positively regulates 
##                53015                 2768                 5191                 2756 
##            regulates 
##                 3216</code></pre>
<p>For other subontologies:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">allm</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/get.html">mget</a></span><span class="op">(</span><span class="va">kk</span>, <span class="va">GOMFCHILDREN</span>, ifnotfound<span class="op">=</span><span class="cn">NA</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">allm</span>, <span class="va">names</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## 
##     isa part of 
##   13631      11</code></pre>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">allcc</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/get.html">mget</a></span><span class="op">(</span><span class="va">kk</span>, <span class="va">GOCCCHILDREN</span>, ifnotfound<span class="op">=</span><span class="cn">NA</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">allcc</span>, <span class="va">names</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## 
##     isa part of 
##    4863    1985</code></pre>
</div>
<div id="dependent-packages" class="section level2">
<h2 class="hasAnchor">
<a href="#dependent-packages" class="anchor"></a>Dependent packages</h2>
<p>Bioconductor software packages that use <code>GO.db</code> are:</p>
<div class="figure">
<img src="bioc_GO.png" alt=""><p class="caption">reverse dependencies</p>
</div>
</div>
</div>
<div id="orgdb-packages-mapping-go-to-genes-via-ncbi" class="section level1">
<h1 class="hasAnchor">
<a href="#orgdb-packages-mapping-go-to-genes-via-ncbi" class="anchor"></a>OrgDb packages – mapping GO to genes via NCBI</h1>
<p><a href="https://bioconductor.org/packages/GO.db/"><code>org.Hs.eg.db</code></a> is a package that uses NCBI mappings between human genes and GO terms to help annotate genes. Here, we use it to obtain all genes directly annotated to a GO term.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>
<span class="va">syms</span> <span class="op">=</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html">select</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span>, keys<span class="op">=</span><span class="va">lk</span>, keytype<span class="op">=</span><span class="st">"GO"</span>, columns<span class="op">=</span><span class="st">"SYMBOL"</span><span class="op">)</span></code></pre></div>
<pre><code>## 'select()' returned 1:many mapping between keys and columns</code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span><span class="va">tt</span>, GO<span class="op">=</span><span class="va">GOID</span><span class="op">)</span>, <span class="va">syms</span>, by<span class="op">=</span><span class="st">"GO"</span><span class="op">)</span>,<span class="fl">15</span><span class="op">)</span></code></pre></div>
<pre><code>##          GOID                             TERM         GO EVIDENCE ONTOLOGY   SYMBOL
## 1  GO:0000001        mitochondrion inheritance GO:0000001     &lt;NA&gt;     &lt;NA&gt;     &lt;NA&gt;
## 2  GO:0000002 mitochondrial genome maintenance GO:0000002      TAS       BP  SLC25A4
## 3  GO:0000002 mitochondrial genome maintenance GO:0000002      IMP       BP     TYMP
## 4  GO:0000002 mitochondrial genome maintenance GO:0000002      ISS       BP    MEF2A
## 5  GO:0000002 mitochondrial genome maintenance GO:0000002      IMP       BP    MPV17
## 6  GO:0000002 mitochondrial genome maintenance GO:0000002      IMP       BP     OPA1
## 7  GO:0000002 mitochondrial genome maintenance GO:0000002      NAS       BP    LONP1
## 8  GO:0000002 mitochondrial genome maintenance GO:0000002      IMP       BP     AKT3
## 9  GO:0000002 mitochondrial genome maintenance GO:0000002      IDA       BP SLC25A36
## 10 GO:0000002 mitochondrial genome maintenance GO:0000002      IEA       BP     PIF1
## 11 GO:0000002 mitochondrial genome maintenance GO:0000002      IDA       BP SLC25A33
## 12 GO:0000002 mitochondrial genome maintenance GO:0000002      IMP       BP    MGME1
## 13 GO:0000003                     reproduction GO:0000003      IBA       BP    GNRH1
## 14 GO:0000003                     reproduction GO:0000003      IBA       BP    GNRH2
## 15 GO:0000003                     reproduction GO:0000003      IEP       BP   MMP23B</code></pre>
<p>To obtain all genes annotated to a GO term OR TO ONE OF ITS CHILD NODES in the GO ontology:</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">syms.all</span> <span class="op">=</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html">select</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span>, keys<span class="op">=</span><span class="va">lk</span>, keytype<span class="op">=</span><span class="st">"GOALL"</span>, columns<span class="op">=</span><span class="st">"SYMBOL"</span><span class="op">)</span></code></pre></div>
<pre><code>## 'select()' returned 1:many mapping between keys and columns</code></pre>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span><span class="va">tt</span>, GOALL<span class="op">=</span><span class="va">GOID</span><span class="op">)</span>, <span class="va">syms.all</span>, by<span class="op">=</span><span class="st">"GOALL"</span><span class="op">)</span>,<span class="fl">15</span><span class="op">)</span></code></pre></div>
<pre><code>##          GOID                             TERM      GOALL EVIDENCEALL ONTOLOGYALL   SYMBOL
## 1  GO:0000001        mitochondrion inheritance GO:0000001        &lt;NA&gt;        &lt;NA&gt;     &lt;NA&gt;
## 2  GO:0000002 mitochondrial genome maintenance GO:0000002         IMP          BP    PARP1
## 3  GO:0000002 mitochondrial genome maintenance GO:0000002         TAS          BP  SLC25A4
## 4  GO:0000002 mitochondrial genome maintenance GO:0000002         IDA          BP     DNA2
## 5  GO:0000002 mitochondrial genome maintenance GO:0000002         IMP          BP     TYMP
## 6  GO:0000002 mitochondrial genome maintenance GO:0000002         IBA          BP     LIG3
## 7  GO:0000002 mitochondrial genome maintenance GO:0000002         ISS          BP    MEF2A
## 8  GO:0000002 mitochondrial genome maintenance GO:0000002         IMP          BP    MPV17
## 9  GO:0000002 mitochondrial genome maintenance GO:0000002         IMP          BP     OPA1
## 10 GO:0000002 mitochondrial genome maintenance GO:0000002         IMP          BP    TOP3A
## 11 GO:0000002 mitochondrial genome maintenance GO:0000002         IEA          BP     TP53
## 12 GO:0000002 mitochondrial genome maintenance GO:0000002         NAS          BP    LONP1
## 13 GO:0000002 mitochondrial genome maintenance GO:0000002         IMP          BP     AKT3
## 14 GO:0000002 mitochondrial genome maintenance GO:0000002         IEA          BP PPARGC1A
## 15 GO:0000002 mitochondrial genome maintenance GO:0000002         IEA          BP    POLG2</code></pre>
<p>Gene to GO mappings can be obtained in a similar fashion also for other model organisms:</p>
<div class="figure">
<img src="orgdb.png" alt=""><p class="caption">OrgDb packages for model organisms</p>
</div>
<p>OrgDb annotation objects for non-model organisms can be obtained via <a href="https://bioconductor.org/packages/AnnotationHub">AnnotationHub</a>.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ah</span> <span class="op">&lt;-</span> <span class="fu">AnnotationHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html">AnnotationHub</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## snapshotDate(): 2021-05-18</code></pre>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">AnnotationHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html">query</a></span><span class="op">(</span><span class="va">ah</span>, <span class="st">"OrgDb"</span><span class="op">)</span></code></pre></div>
<pre><code>## AnnotationHub with 1744 records
## # snapshotDate(): 2021-05-18
## # $dataprovider: ftp://ftp.ncbi.nlm.nih.gov/gene/DATA/
## # $species: Escherichia coli, greater Indian_fruit_bat, Zythia rabiei, Zymoseptoria tritici_IPO3...
## # $rdataclass: OrgDb
## # additional mcols(): taxonomyid, genome, description, coordinate_1_based, maintainer,
## #   rdatadateadded, preparerclass, tags, rdatapath, sourceurl, sourcetype 
## # retrieve records with, e.g., 'object[["AH92572"]]' 
## 
##             title                                     
##   AH92572 | org.Ag.eg.db.sqlite                       
##   AH92573 | org.At.tair.db.sqlite                     
##   AH92574 | org.Bt.eg.db.sqlite                       
##   AH92575 | org.Cf.eg.db.sqlite                       
##   AH92576 | org.Gg.eg.db.sqlite                       
##   ...       ...                                       
##   AH95560 | org.Linderina_pennispora.eg.sqlite        
##   AH95561 | org.Daldinia_childiae.eg.sqlite           
##   AH95562 | org.Aspergillus_nidulans_FGSC_A4.eg.sqlite
##   AH95563 | org.Emericella_nidulans_FGSC_A4.eg.sqlite 
##   AH95564 | org.Talaromyces_atroroseus.eg.sqlite</code></pre>
</div>
<div id="biocview-go-used-for-73-packages-in-bioc-3-13" class="section level1">
<h1 class="hasAnchor">
<a href="#biocview-go-used-for-73-packages-in-bioc-3-13" class="anchor"></a>biocView: GO used for 73 packages in Bioc 3.13</h1>
<p>Bioconductor package contributors are required to provide tags for their packages derived from a vocabulary called <code>biocViews</code>. “GO” is a tag and 73 packages in Bioconductor 3.13 use it to describe their objective.</p>
<div class="figure">
<img src="GOview.png" alt=""><p class="caption">biocView GO</p>
</div>
<p>Filter the landing pages for Bioc 3.13 packages self-describing with the ‘GO’ tag <a href="https://bioconductor.org/packages/3.13/BiocViews.html#___GO">here</a>.</p>
<p>For Bioconductor 3.14 packages in development, see <a href="https://bioconductor.org/packages/3.14/BiocViews.html#___GO">here</a>.</p>
</div>
<div id="go-enrichment-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#go-enrichment-analysis" class="anchor"></a>GO enrichment analysis</h1>
<p>A popular use case for GO annotations is GO enrichment analysis. A variety of methods are available in Bioconductor for that purpose, and can typically be divided into (i) methods that treat GO annotations as gene sets, and (ii) methods that account for the topology of the GO graph.</p>
<div id="methods-that-treat-go-annotations-as-gene-sets" class="section level2">
<h2 class="hasAnchor">
<a href="#methods-that-treat-go-annotations-as-gene-sets" class="anchor"></a>Methods that treat GO annotations as gene sets</h2>
<p>Among the most used representatives of this class of methods are</p>
<ul>
<li>
<code><a href="https://rdrr.io/pkg/limma/man/goana.html">limma::goana</a></code> which performs classic over-representation analysis using a hypergeometric test as originally developed for the analysis of microarray data, and</li>
<li>
<a href="https://bioconductor.org/packages/goseq">GOSeq</a> which conducts enrichment testing of GO terms while adjusting for gene length bias in RNA-seq data.</li>
</ul>
</div>
<div id="methods-that-account-for-the-topology-of-the-go-graph" class="section level2">
<h2 class="hasAnchor">
<a href="#methods-that-account-for-the-topology-of-the-go-graph" class="anchor"></a>Methods that account for the topology of the GO graph</h2>
<p>Among the most used representatives of this class of methods are</p>
<ul>
<li>
<a href="https://bioconductor.org/packages/GOstats">GOstats</a> which performs a hypergeometric test that optionally conditions on the GO structure, and</li>
<li>
<a href="https://bioconductor.org/packages/topGO">topGO</a> which implements different approaches (elim, weight, and parentchild) to reduce redundancy resulting from genes being annotated along the parent-child hierarchy.</li>
</ul>
</div>
</div>
<div id="benchmarking-of-methods-for-go-enrichment-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#benchmarking-of-methods-for-go-enrichment-analysis" class="anchor"></a>Benchmarking of methods for GO enrichment analysis</h1>
<p>The <a href="https://bioconductor.org/packages/GSEABenchmarkeR">GSEABenchmarkeR</a> package ships disease relevance scores for GO-BP and GO-MF terms. These relevance scores are aggregated per-gene disease relevance scores as obtained from the <a href="https://www.malacards.org/">MalaCards</a> database. This establishes some ground truth in a benchmarking setup, indicating which GO terms are considered relevant for a certain disease.</p>
<p>As an example, we inspect relevance scores of certain GO-BP terms for breast cancer:</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">data.dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package<span class="op">=</span><span class="st">"GSEABenchmarkeR"</span><span class="op">)</span>
<span class="va">mala.gobp.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data.dir</span>, <span class="st">"malacards"</span>, <span class="st">"GO_BP.rds"</span><span class="op">)</span>
<span class="va">mala.gobp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="va">mala.gobp.file</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">mala.gobp</span><span class="op">$</span><span class="va">BRCA</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>##                                                                    TITLE REL.SCORE MATCHED.GENES
## GO:0008284                     Positive Regulation of Cell Proliferation    166.10            78
## GO:0048010 Vascular Endothelial Growth Factor Receptor Signaling Pathway    135.71            56
## GO:0043066                      Negative Regulation of Apoptotic Process    123.02            67
## GO:0001934                Positive Regulation of Protein Phosphorylation    111.96            40
## GO:0007173            Epidermal Growth Factor Receptor Signaling Pathway    111.09            52
## GO:0048015                       Phosphatidylinositol-mediated Signaling    110.84            37
##            TOTAL.GENES
## GO:0008284         491
## GO:0048010         281
## GO:0043066         530
## GO:0001934         151
## GO:0007173         323
## GO:0048015         121</code></pre>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.1.0 RC (2021-05-10 r80294)
## Platform: aarch64-apple-darwin20 (64-bit)
## Running under: macOS Big Sur 11.4
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/4.1-arm64/Resources/lib/libRblas.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/4.1-arm64/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats4    parallel  stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] org.Hs.eg.db_3.13.0  dplyr_1.0.7          GO.db_3.13.0         AnnotationDbi_1.54.1
##  [5] IRanges_2.26.0       S4Vectors_0.30.0     Biobase_2.52.0       AnnotationHub_3.1.5 
##  [9] BiocFileCache_2.0.0  dbplyr_2.1.1         BiocGenerics_0.38.0 
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_1.0.7                    Biostrings_2.60.2             png_0.1-7                    
##  [4] assertthat_0.2.1              rprojroot_2.0.2               digest_0.6.28                
##  [7] utf8_1.2.2                    mime_0.11                     GenomeInfoDb_1.28.4          
## [10] R6_2.5.1                      RSQLite_2.2.8                 evaluate_0.14                
## [13] httr_1.4.2                    pillar_1.6.3                  zlibbioc_1.38.0              
## [16] rlang_0.4.11                  curl_4.3.2                    jquerylib_0.1.4              
## [19] blob_1.2.2                    rmarkdown_2.11                pkgdown_1.6.1                
## [22] textshaping_0.3.5             desc_1.3.0                    stringr_1.4.0                
## [25] RCurl_1.98-1.5                bit_4.0.4                     shiny_1.7.0                  
## [28] compiler_4.1.0                httpuv_1.6.3                  xfun_0.26                    
## [31] pkgconfig_2.0.3               systemfonts_1.0.2             htmltools_0.5.2              
## [34] KEGGREST_1.32.0               tidyselect_1.1.1              GenomeInfoDbData_1.2.6       
## [37] tibble_3.1.4                  interactiveDisplayBase_1.31.1 fansi_0.5.0                  
## [40] withr_2.4.2                   crayon_1.4.1                  later_1.3.0                  
## [43] bitops_1.0-7                  rappdirs_0.3.3                jsonlite_1.7.2               
## [46] xtable_1.8-4                  lifecycle_1.0.1               DBI_1.1.1                    
## [49] magrittr_2.0.1                stringi_1.7.4                 cachem_1.0.6                 
## [52] XVector_0.32.0                fs_1.5.0                      promises_1.2.0.1             
## [55] bslib_0.3.0                   ellipsis_0.3.2                filelock_1.0.2               
## [58] ragg_1.1.3                    generics_0.1.0                vctrs_0.3.8                  
## [61] tools_4.1.0                   bit64_4.0.5                   glue_1.4.2                   
## [64] purrr_0.3.4                   BiocVersion_3.13.1            fastmap_1.1.0                
## [67] yaml_2.2.1                    BiocManager_1.30.16           memoise_2.0.0                
## [70] knitr_1.34                    sass_0.4.0</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Vince Carey.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
